llama_stack/providers/inline/vector_io/sqlite_vec/sqlite_vec.py

```diff
@@ -71,30 +71,33 @@ async def delete(self):
 
     async def add_chunks(self, chunks: List[Chunk], embeddings: NDArray):
         """
-        Add new chunks along with their embeddings.
+        Add new chunks along with their embeddings using batch inserts.
         For each chunk, we insert its JSON into the metadata table and then insert its
         embedding (serialized to raw bytes) into the virtual table using the assigned rowid.
-        If any insert fails, the transaction is rolled back to maintain consistency.
+        If any insert fails, the transaction is rolled back.
         """
         cur = self.connection.cursor()
         try:
             # Start transaction
             cur.execute("BEGIN TRANSACTION")
-            for chunk, emb in zip(chunks, embeddings, strict=False):
-                # Serialize and insert the chunk metadata.
-                chunk_json = chunk.model_dump_json()
-                cur.execute(f"INSERT INTO {self.metadata_table} (chunk) VALUES (?)", (chunk_json,))
-                row_id = cur.lastrowid
-                # Ensure the embedding is a list of floats.
-                emb_list = emb.tolist() if isinstance(emb, np.ndarray) else list(emb)
-                emb_blob = serialize_vector(emb_list)
-                cur.execute(f"INSERT INTO {self.vector_table} (rowid, embedding) VALUES (?, ?)", (row_id, emb_blob))
+            # Serialize and insert the chunk metadata.
+            chunk_data = [(chunk.model_dump_json(),) for chunk in chunks]
+            cur.executemany(f"INSERT INTO {self.metadata_table} (chunk) VALUES (?)", chunk_data)
+            # Fetch the last *n* inserted row_ids -- note: this is more reliable than `row_id = cur.lastrowid`
+            cur.execute(f"SELECT rowid FROM {self.metadata_table} ORDER BY rowid DESC LIMIT {len(chunks)}")
```

Comment:
instead of relying on row-ids which are implicit like this, can we augment the chunks with identifiers instead? that way you can break this into two transactions also.

Reply:
Yeah, of course. I'll start that today.

Reply:
Thanks again @ashwinb. Feel free to take a look when you have a second.

Reply:
@ashwinb any chance you can take a second pass? Made some updates.